annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastEnvironment.java

```diff
@@ -55,7 +75,38 @@ public static FastEnvironment build(RoundEnvironment environment, Elements eleme
         void validate() {
             validatePackageNames();
             validatePackageNameDuplicates();
-            getFastPackages().forEach(it -> validateTypeNameDuplicates(it));
+            getFastPackages().forEach(it -> {
+                validateTypeNameDuplicates(it);
+                it.getFastTypes().forEach(fastType -> {
+                    validateDefaultConstructor(fastType);
+                    validateFastFieldDuplicates(fastType.getFastFields());
+                });
+            });
+
+        }
+
+        private void validateDefaultConstructor(FastTypeElement fastType) {
+            for (Element childElement : fastType.getElement().getEnclosedElements()) {
+                if (childElement.getKind() == ElementKind.CONSTRUCTOR) {
```

Comment:
Type may have multiple constructors. You'll report any non-default constructor as error. Your code doesn't check if class is abstract. Inner classes shouldn't be allowed as well.
Move this check to another branch and PR. There is separate ticket for this work

Reply:
This code should be in separate PR